waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -550,8 +550,9 @@
     {:failed-instances (vals (get @service-id->failed-instances-transient-store service-id))
      :syncer (retrieve-syncer-state-fn service-id)})
 
-  (state [_]
-    {:service-id->failed-instances @service-id->failed-instances-transient-store
+  (state [{:keys [watch-state]}]
+    {:watch-state @watch-state
```

Comment:
We should have an integration tests that verifies a sufficiently narrow expected modification in the watch state.